File: src/main/java/com/puhovin/intellijplugin/twm/util/ToolWindowManagerBundle.java
package com.puhovin.intellijplugin.twm.util;

import com.intellij.AbstractBundle;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.PropertyKey;

import java.lang.ref.Reference;
import java.lang.ref.SoftReference;
import java.util.ResourceBundle;

public final class ToolWindowManagerBundle {

    @NonNls
    private static final String BUNDLE = "com.puhovin.intellijplugin.twm.bundle";
    private static Reference<ResourceBundle> ourBundle = null;

    private ToolWindowManagerBundle() {}

    private static ResourceBundle getBundle() {
        ResourceBundle bundle = null;

        if (ourBundle != null) {
            bundle = ourBundle.get();
        }

        if (bundle == null) {
            bundle = ResourceBundle.getBundle(BUNDLE);
            ourBundle = new SoftReference<>(bundle);
        }

        return bundle;
    }

    public static String message(@PropertyKey(resourceBundle = BUNDLE) final String key, final Object... params) {
        return AbstractBundle.message(getBundle(), key, params);
    }
}

File: src/main/kotlin/com/puhovin/intellijplugin/twm/util/ToolWindowManagerBundle.kt
package com.puhovin.intellijplugin.twm.util

import com.intellij.AbstractBundle
import java.util.ResourceBundle
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey

object ToolWindowManagerBundle {
    @NonNls
    private const val BUNDLE_PATH = "com.puhovin.intellijplugin.twm.bundle"
    private val bundle: ResourceBundle by lazy { ResourceBundle.getBundle(BUNDLE_PATH) }

    fun message(@PropertyKey(resourceBundle = BUNDLE_PATH) key: String, vararg params: Any): String {
        return AbstractBundle.message(bundle, key, *params)
    }
}
